src/panels/config/logs/error-log-card.ts

```diff
@@ -42,52 +50,136 @@ class ErrorLogCard extends LitElement {
 
   @property({ type: Boolean, attribute: true }) public show = false;
 
-  @state() private _isLogLoaded = false;
+  @query(".error-log") private _logElement?: HTMLElement;
+
+  @query("#scroll-marker") private _scrollMarkerElement?: HTMLElement;
+
+  @query("#nr-of-lines-menu") private _nrOfLinesMenuElement?: HaMenu;
+
+  @state() private _logs: string[] = [];
+
+  @state() private _logHTML?: string | TemplateResult[];
+
+  @state() private _scrolledToBottomController =
+    new IntersectionController<boolean>(this, {
+      callback(this: IntersectionController<boolean>, entries) {
+        return entries[0].isIntersecting;
+      },
+    });
+
+  @state() private _newLogsIndicator?: boolean;
 
-  @state() private _logHTML?: TemplateResult[] | TemplateResult | string;
+  @state() private _numberOfLines = NUMBER_OF_LINES_OPTIONS[0];
 
   @state() private _error?: string;
 
+  @state() private _logStreamAborter?: AbortController;
+
   protected render(): TemplateResult {
+    let filteredLines = this._logHTML;
+
+    if (Array.isArray(this._logHTML) && this._logHTML.length && this.filter) {
+      filteredLines = this._logHTML.filter((_line, key) =>
+        this._logs[key]
+          .toLocaleLowerCase()
+          .includes(this.filter.toLocaleLowerCase())
+      );
+    }
+
     return html`
       <div class="error-log-intro">
         ${this._error
           ? html`<ha-alert alert-type="error">${this._error}</ha-alert>`
           : ""}
-        ${this._logHTML
-          ? html`
-              <ha-card outlined>
-                <div class="header">
-                  <h1 class="card-header">
-                    ${this.header ||
-                    this.hass.localize("ui.panel.config.logs.show_full_logs")}
-                  </h1>
+        <ha-card
+          outlined
+          class=${classMap({ hidden: this._logHTML === undefined })}
```

Comment:
Just conditionally render `ha-card` like before instead of hiding it with css

Reply:
I have done it like this that the observer works, otherwise I would need to unobserve before removing ha-card and re observere when it comes back.